utils/env_preferences/src/linux.rs

```diff
@@ -49,6 +49,9 @@ impl FromStr for LocaleCategory {
     }
 }
 
+// TODO: Add a function to return all the locales POSIX categories explicitly
+
+/// Retrieves locale for `LC_ALL` POSIX category. Returns other categories unless explicitly set in the thread
```

Comment:
I don't understand what this means

Reply:
Actually, when we select `setlocale` which the posix locale type (for eg in this case it is LC_ALL), it only retrieves that locales. 

Here my belief is that users should have option to retrieve all the types (for example LC_ADDRESS etc). The intial thought not to add them was having to call `unsafe` functions, I am much comfortable with `unsafe` as compared to initially starting this project

Reply:
so does this method return all locale categories?

Reply:
will*, I can switch it to return to categories in this PR itself

Reply:
I'm not suggesting any changes to the code, I just still don't understand the comment. What does "Returns other categories unless explicitly set in the thread" *mean*? Can you rephrase so I can understand?

Reply:
Like when I testing this function, I noticed that if locale is set and then retrieved the response it something like:
```rust
// Set the locale
setlocale(LC_ADDRESS, 'de_CH');
// Now retrieving the locale for LC_ALL category
let resp = setlocale(LC_ALL, null)
```

Now, this resp will consist of all the categories. BUT if I do this 

```rust
let resp = setlocale(LC_ALL, null)
```

I will only recieve a single locale.

This has been done to prevent panics

Reply:
So should it say "Also returns other categories if any are explicitly set in the thread"? I think the use of "unless" is incorrect following your explanation.